tests/conftest.py

```diff
@@ -214,6 +214,28 @@ def gordo_ml_server_client(
             yield app.test_client()
 
 
+@pytest.fixture()
+def postgresdb():
+    client = docker.from_env()
+
+    postgres = None
+    try:
+        postgres = client.containers.run(
+            image="postgres:11-alpine",
+            environment={"POSTGRES_USER": "postgres", "POSTGRES_PASSWORD": "postgres"},
+            ports={"5432/tcp": "5432"},
+            remove=True,
+            detach=True,
+        )
+        import time
+
+        time.sleep(5)
+        yield
+    finally:
+        if postgres is not None:
+            postgres.kill()
```

Comment:
This looks like it might work the same without the try/finally.

Reply:
That's true if decorated with [@pytest.yield_fixture](https://pytest.readthedocs.io/en/2.8.7/yieldfixture.html), which this should probably be, but it'll be functionally the same. As it is now, a `KeyboardInterrupt` wouldn't guarantee the container would be killed; but from the docs it seems `yield_fixture` would in the same way... :+1: